raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -1477,6 +1477,7 @@ private void appendBatch(
     ) {
         try {
             int epoch = state.epoch();
+            batch.data.batches().forEach(recordBatch -> recordBatch.setPartitionLeaderEpoch(epoch));
```

Comment:
This is necessary for the sanity check in `Log.scala append()`
```
         if (batch.magic >= RecordBatch.MAGIC_VALUE_V2) {
              maybeAssignEpochStartOffset(batch.partitionLeaderEpoch, batch.baseOffset)
```

Reply:
Hmm.. Do we need this? I thought we already set leader epoch here: https://github.com/apache/kafka/blob/trunk/raft/src/main/java/org/apache/kafka/raft/internals/BatchBuilder.java#L256.

Reply:
@hachikuji I thought the same as you, re-check the code, the `PARTITION_LEADER_EPOCH_OFFSET` has been set in `DefaultRecordBatch#writeHeader`, but the problem is that the `BatchBuilder` is always constructed with the `epoch=RecordBatch.NO_PARTITION_LEADER_EPOCH`, so I updated it and reverted the L1480.